"TypeError: wasm function signature contains illegal type" error on current dev branch

Running `make test` using Go 1.21 using the latest `dev` branch results in errors:

```
tinygo:ld.lld: error: undefined symbol: notARealFunction                                                                                                                     
>>> referenced by builderr.go:9 (/home/ron/Development/tinygo/tinygo/tests/testing/builderr/builderr.go:9)                                                                   
>>>               /home/ron/.cache/tinygo/thinlto/llvmcache-4457084472EE654FE0B37BA07A9F7F0BFC5D7957:(github.com/tinygo-org/tinygo/tests/testing/builderr_test.TestThing)    
failed to run tool: ld.lld                                                                                                                                                   
TypeError: wasm function signature contains illegal type                                                                                                                     
    at (syscall/js.Value).Get (<anonymous>:wasm-function[142]:0x12a0a)                                                                                                       
    at runtime.run$1 (<anonymous>:wasm-function[180]:0x18dc8)                                                                                                                
    at runtime.run$1$gowrapper (<anonymous>:wasm-function[130]:0x111bc)                                                                                                      
    at tinygo_launch (<anonymous>:wasm-function[14]:0x5f7)                                                                                                                   
    at (*internal/task.Task).Resume (<anonymous>:wasm-function[77]:0x8ed2)                                                                                                   
    at runtime.scheduler (<anonymous>:wasm-function[131]:0x11274)                                                                                                            
    at _start (<anonymous>:wasm-function[128]:0x110e7)
    at _start.command_export (<anonymous>:wasm-function[415]:0x579f4)
    at global.Go.run (/home/ron/Development/tinygo/tinygo/targets/wasm_exec.js:480:24) 
    at /home/ron/Development/tinygo/tinygo/targets/wasm_exec.js:523:14
TypeError: wasm function signature contains illegal type
    at (syscall/js.Value).Get (<anonymous>:wasm-function[142]:0x12a0c)
    at runtime.run$1 (<anonymous>:wasm-function[180]:0x18dca)
    at runtime.run$1$gowrapper (<anonymous>:wasm-function[130]:0x111be)
    at tinygo_launch (<anonymous>:wasm-function[14]:0x5f9)
    at (*internal/task.Task).Resume (<anonymous>:wasm-function[77]:0x8ed4)
    at runtime.scheduler (<anonymous>:wasm-function[131]:0x11276)
    at _start (<anonymous>:wasm-function[128]:0x110e9)
    at _start.command_export (<anonymous>:wasm-function[417]:0x5833f)
    at global.Go.run (/home/ron/Development/tinygo/tinygo/targets/wasm_exec.js:480:24) 
    at /home/ron/Development/tinygo/tinygo/targets/wasm_exec.js:523:14
--- FAIL: TestTest (0.00s)
    --- FAIL: TestTest/WASM (0.00s)
        --- FAIL: TestTest/WASM/Pass (19.98s)
            main_test.go:531: FAIL      github.com/tinygo-org/tinygo/tests/testing/pass 1.107s
            main_test.go:443: test failed
```

My OS:

```
$ tinygo version
tinygo version 0.29.0-dev-f4375d04 linux/amd64 (using go version go1.21.0 and LLVM version 15.0.0)
```

--- a/Makefile
+++ b/Makefile
@@ -267,12 +267,22 @@ lib/wasi-libc/sysroot/lib/wasm32-wasi/libc.a:
 	@if [ ! -e lib/wasi-libc/Makefile ]; then echo "Submodules have not been downloaded. Please download them using:\n  git submodule update --init"; exit 1; fi
 	cd lib/wasi-libc && make -j4 EXTRA_CFLAGS="-O2 -g -DNDEBUG -mnontrapping-fptoint -msign-ext" MALLOC_IMPL=none CC=$(CLANG) AR=$(LLVM_AR) NM=$(LLVM_NM)
 
+# Check for Node.js used during WASM tests.
+NODEJS_VERSION := $(word 1,$(subst ., ,$(shell node -v | cut -c 2-)))
+MIN_NODEJS_VERSION=16
+
+.PHONY: check-nodejs-version
+check-nodejs-version:
+ifeq (, $(shell which node))
+	@echo "Install NodeJS version 16+ to run tests."; exit 1;
+endif
+	@if [ $(NODEJS_VERSION) -lt $(MIN_NODEJS_VERSION) ]; then echo "Install NodeJS version 16+ to run tests."; exit 1; fi
 
 # Build the Go compiler.
 tinygo:
 	@if [ ! -f "$(LLVM_BUILDDIR)/bin/llvm-config" ]; then echo "Fetch and build LLVM first by running:"; echo "  make llvm-source"; echo "  make $(LLVM_BUILDDIR)"; exit 1; fi
 	CGO_CPPFLAGS="$(CGO_CPPFLAGS)" CGO_CXXFLAGS="$(CGO_CXXFLAGS)" CGO_LDFLAGS="$(CGO_LDFLAGS)" $(GOENVFLAGS) $(GO) build -buildmode exe -o build/tinygo$(EXE) -tags "byollvm osusergo" -ldflags="-X github.com/tinygo-org/tinygo/goenv.GitSha1=`git rev-parse --short HEAD`" .
-test: wasi-libc
+test: wasi-libc check-nodejs-version
 	CGO_CPPFLAGS="$(CGO_CPPFLAGS)" CGO_CXXFLAGS="$(CGO_CXXFLAGS)" CGO_LDFLAGS="$(CGO_LDFLAGS)" $(GO) test $(GOTESTFLAGS) -timeout=20m -buildmode exe -tags "byollvm osusergo" ./builder ./cgo ./compileopts ./compiler ./interp ./transform .
 
 # Standard library packages that pass tests on darwin, linux, wasi, and windows, but take over a minute in wasi

